messaging/src/main/java/org/axonframework/commandhandling/gateway/ExponentialBackOffIntervalRetryScheduler.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2010-2019. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandMessage;
+
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static org.axonframework.common.BuilderUtils.assertPositive;
+
+/**
+ * A RetryScheduler that uses a backoff strategy, retrying commands at increasing intervals when they fail because of
+ * an exception that is not explicitly non-transient. Checked exceptions are considered non-transient and will not
+ * result in a retry.
+ *
+ * @author Bert Laverman &lt;bert.laverman@axoniq.io&gt;
+ * @since 4.2
+ */
+public class ExponentialBackOffIntervalRetryScheduler extends AbstractRetryScheduler {
+
+    private static final long DEFAULT_BACKOFF_FACTOR = 100L;
+
+    private final long backoffFactor;
+
+    /**
+     * Instantiate an {@link ExponentialBackOffIntervalRetryScheduler}. the settings are copied from the
+     * {@link Builder} and validated.
+     *
+     * @param builder
+     */
+    private ExponentialBackOffIntervalRetryScheduler(Builder builder) {
+        super(builder);
+        builder.validate();
```

Comment:
The `builder.valdiate()` call could be moved to the `AbstractRetryScheduler`, thus ensuring anyone creating an implementation of it will have at least the `retryExecutor` and `maxRetryCount` validated.

Reply:
While `validate()` is also overridden in extensions? this means a call in the abstract base class will cause validations in extensions? I'd prefer to have the moment of validation controlled by the concrete class.

Reply:
Exactly, behavior an implementation can override by providing their own `validate()` method, which does not call the parent `Builder`'s `validate()` method. 

Any how, I'd personally prefer conciseness towards all the others Builder approaches, which do exactly as I've stated. Added, I think there's a way around your comment too.

@m1l4n54v1c, do you want to be the judge about this?

Reply:
`validate` should be invoked in a superclass. With this approach, we are invoking validate twice. I agree with the approach we're already having.

Reply:
Ok, ok. Removed all calls except one in the `AbstractRetryScheduler`'s constructor.